modules/parquet/src/parquetjs/reader.ts

```diff
@@ -576,17 +635,23 @@ function decodeDataPage(
   }
   let valueCountNonNull = 0;
   for (const dlvl of dLevels) {
-    if (dlvl === column.dLevelMax) {
+    if (dlvl === opts.column.dLevelMax) {
       valueCountNonNull++;
     }
   }
 
   /* read values */
   const valueEncoding = getThriftEnum(Encoding, header.data_page_header?.encoding!) as ParquetCodec;
-  const values = decodeValues(column.primitiveType!, valueEncoding, dataCursor, valueCountNonNull, {
-    typeLength: column.typeLength,
-    bitWidth: column.typeLength
-  });
+  const values = decodeValues(
+    opts.column.primitiveType!,
+    valueEncoding,
+    dataCursor,
+    valueCountNonNull,
+    {
```

Comment:
Maybe introduce a helper variable `decodeOptions` to make this call easier to read?

Reply:
Done